test/test_examples.py

```diff
@@ -1297,5 +1298,90 @@ def strptime(s):
         # End Versioned API Example 8
 
 
+class TestSnapshotQueryExamples(IntegrationTest):
+    @client_context.require_replica_set
```

Comment:
Why doesn't this test work on sharded clusters?

Reply:
From this [failure](https://spruce.mongodb.com/task/mongo_python_driver_tests_python_version_ubuntu18_test_ssl__platform~ubuntu_18.04_auth_ssl~auth_ssl_python_version~3.10_coverage~coverage_test_5.0_standalone_patch_861d79537fee9dd80be02a4ffdbadc5897acfd7d_623dd29ca4cf47697943fc09_22_03_25_14_33_01/tests?execution=0&sortBy=STATUS&sortDir=ASC):

`pymongo.errors.OperationFailure: node needs to be a replica set member to use readConcern: snapshot, full error: {'ok': 0.0, 'errmsg': 'node needs to be a replica set member to use readConcern: snapshot', 'code': 123, 'codeName': 'NotAReplicaSet'}`

Reply:
Oh I see, either one should work

Reply:
Updated